src/popover/Popover.js

```diff
@@ -1,99 +1,308 @@
+import cx from 'classnames';
 
 const Popover = React.createClass({
 
   propTypes: {
-    children: React.PropTypes.oneOfType([
-      React.PropTypes.node,
-      React.PropTypes.element,
-      React.PropTypes.array
-    ]).isRequired,
-    content: React.PropTypes.oneOfType([
-      React.PropTypes.node,
-      React.PropTypes.element,
-      React.PropTypes.array
-    ]).isRequired,
-    isOpen: React.PropTypes.bool.isRequired,
-    popoverClassName: React.PropTypes.string
+    children: React.PropTypes.node.isRequired,
+    popover: React.PropTypes.shape({
+      content: React.PropTypes.node.isRequired,
+      type: React.PropTypes.oneOf(['absolute', 'relative']),
+      position: React.PropTypes.oneOf(['top', 'bottom']),
+      anchor: React.PropTypes.oneOf(['left', 'center', 'right']),
+      event: React.PropTypes.oneOf(['click', 'hover']),
+      onShow: React.PropTypes.func,
+      onHide: React.PropTypes.func,
+      dismissOnScroll: React.PropTypes.bool,
+      className: React.PropTypes.string,
+      id: React.PropTypes.string,
+      maxWidth: React.PropTypes.oneOfType([
+        React.PropTypes.number,
+        React.PropTypes.string
+      ])
+    }).isRequired,
+    id: React.PropTypes.string,
+    className: React.PropTypes.string,
+    style: React.PropTypes.object
   },
 
   getDefaultProps() {
     return {
+      id: '',
       className: '',
-      popoverClassName: '',
-      style: {},
-      onVisible: () => {},
-      onHidden: () => {}
+      style: {}
     };
   },
 
   getInitialState() {
     return {
-      visible: false
+      isOpen: false
     };
   },
 
   componentDidMount() {
-    const userSettings = omit(this.props, omittedProps);
-    const settings = {
-      on: 'manual',
-      popup: $(this.refs.popover.getDOMNode()),
-      onVisible: this.onVisible,
-      onHidden: this.onHidden,
-      hideOnScroll: false // TODO
+    this.saveValuesFromNodeTree();
+  },
+
+  saveValuesFromNodeTree(cb) {
+    const childrenNode = this.refs.children.getDOMNode();
+    const popoverNode = this.isAbsolute() ? this.popoverNode : childrenNode.childNodes[1];
+
+    if (popoverNode) {
+      const { clientWidth: childWidth, clientHeight: childHeight } = childrenNode;
+      const { clientHeight: popoverHeight, clientWidth: popoverWidth } = popoverNode;
+      const { top: offsetTop, left: offsetLeft } = this.getOffsetRect();
+      this.setState({
+        childWidth,
+        childHeight,
+        popoverWidth,
+        popoverHeight,
+        offsetTop,
+        offsetLeft
+      }, cb);
+    }
+  },
+
+  // extend with default values
+  getPopoverProps() {
+    return {
+      ...{
+        type: 'relative',
+        position: 'top',
+        anchor: 'center',
+        event: 'hover',
+        onShow: () => {},
+        onHide: () => {},
+        dismissOnScroll: true,
+        className: ''
+      },
+      ...this.props.popover
```

Comment:
`{ ...{ prop1: value1, prop2: value2 }, ...rest }` should be the same of `{ prop1: value1, prop2: value2 , ...rest }`